.github/workflows/lint.yml

```diff
@@ -57,3 +57,28 @@ jobs:
     - name: "run cabal-fmt"
       run: |
         cabal-fmt --check spectrometer.cabal
+
+  cabal-install-check:
+    name: cabal-install-check
+    runs-on: ubuntu-latest
+    container: ghcr.io/fossas/haskell-dev-tools:9.0.2
+
+    steps:
+    - uses: actions/checkout@v3
+
+    # Run `cabal install`.
+    - uses: actions/cache@v2
+      name: Cache cabal store
+      with:
+        path: ${{ steps.setup-haskell.outputs.cabal-store || '~/.cabal/store' }}
+        key: ${{ runner.os }}-${{ matrix.ghc }}-cabal-cache-${{ hashFiles('**/*.cabal') }}
+        restore-keys: |
+          ${{ runner.os }}-${{ matrix.ghc }}-cabal-cache-
+          ${{ runner.os }}-${{ matrix.ghc }}-
+          ${{ runner.os }}-
+
+    - name: Run `cabal install`
+      run: |
+        apk add xz-dev bzip2-dev bzip2-static
+        cabal update
+        cabal install
```

Comment:
`--dry-run` isn't sufficent here?  It might be faster and it avoids modifying the container.

Reply:
`--dry-run` is insufficient because we need to actually execute the build in order to see things like Template Haskell compilation errors.